### PR TITLE
Several issues and bug fixes

### PR DESCRIPTION
**Pull request by cemsbr**
This pull request fixes many bugs and issues. During development, it was difficult to keep several pull requests in different branches, so it also includes other 2 pull requests (#133, #137).

### CODE REVIEW

File: pyof/v0x01/foundation/basic_types.py [hunk 1]
```diff
@@ -201,6 +201,14 @@ def get_size(self):
         return 6
 
 
+def _int2hex(n):
```
Comment (beraldoleal) [inline pyof/v0x01/foundation/basic_types.py]: This should be inside the method unpack() or a `util.py` file. I will accept the PR but, please open an issue for that.

Comment (cemsbr) [inline pyof/v0x01/foundation/basic_types.py]: Created #143.

File: pyof/v0x01/foundation/base.py [hunk 2]
```diff
@@ -110,7 +110,7 @@ def value(self):
         else:
             return self._value
 
-    def pack(self):
+    def pack(self, value=None):
         """Pack the value as a binary representation.
 
         Returns:
```
Comment (beraldoleal) [inline pyof/v0x01/foundation/base.py]: We need to document the value argument.

File: pyof/v0x01/foundation/basic_types.py [hunk 3]
```diff
@@ -485,16 +527,31 @@ def get_size(self):
...
         """
+        if isinstance(value, type(self)):
+            return value.pack()
+
+        if value is None:
+            value = self
+        else:
+            container = type(self)()
+            container.extend(value)
+            value = value
```
Comment (beraldoleal) [inline pyof/v0x01/foundation/basic_types.py]: value = container ?

**Comment by beraldoleal** [comment]
Thanks for work on this. I will wait for the fixes to accept the merge.

**Comment by diraol** [comment]
@beraldoleal I think that I've managed all questions you noted here. =)

**Comment by beraldoleal** [comment]
Thanks @diraol .

File: pyof/v0x01/foundation/basic_types.py (hunk 3 above)
Comment (cemsbr) [inline pyof/v0x01/foundation/basic_types.py]: This is already fixed in #141.

**Pull request merged**